src/clarity-icons/clarity-icons.spec.ts

```diff
@@ -373,15 +356,134 @@ describe("ClarityIcons", () => {
             expect(clarityIcon.style.width).toBe("25px");
             expect(clarityIcon.style.height).toBe("25px");
         });
+
+        it("should be able to use non-svg template", () => {
+            const testShape = `<img src="arrow-icon.png" alt="arrow-icon" width="42" height="42">`;
+            ClarityIcons.add({"test-shape": testShape});
+
+            const clarityIcon = document.createElement("clr-icon");
+            clarityIcon.setAttribute("shape", "test-shape");
+
+            expect(removeWhitespace(clarityIcon.innerHTML)).toBe(removeWhitespace(testShape));
+        });
+
+        it("should not change non-svg template if title attribute is specified", () => {
+            const testShape = `<img src="arrow-icon.png" alt="arrow-icon" width="42" height="42">`;
+            ClarityIcons.add({"test-shape": testShape});
+
+            const clarityIcon = document.createElement("clr-icon");
+            clarityIcon.setAttribute("shape", "test-shape");
+            clarityIcon.setAttribute("title", "my-custom-title");
+
+            expect(removeWhitespace(clarityIcon.innerHTML)).toBe(removeWhitespace(testShape));
+        });
+
+        it("should not inject anything if the custom title is not given", () => {
+            const customTitle = "my-custom-title";
+            const testShape = `<svg><g></g></svg>`;
+            ClarityIcons.add({"test-shape": testShape});
+
+            const clarityIcon = document.createElement("clr-icon");
+            clarityIcon.setAttribute("shape", "test-shape");
+
+            expect(clarityIcon.hasAttribute("aria-labelledby")).toBeFalsy();
+            expect(removeWhitespace(clarityIcon.innerHTML)).toBe(removeWhitespace(testShape));
+        });
+
+        it("should inject a custom title into the template if title attribute is present", () => {
+            const timeStart = performance.now();
+            const clarityIcon = document.createElement("clr-icon");
+            const customTitle = "my-custom-title";
+
+            clarityIcon.setAttribute("shape", "angle");
+            const timeAngleShape = performance.now();
+            console.log("Icon load time: " + (timeAngleShape - timeStart));
+
+            clarityIcon.setAttribute("title", customTitle);
+            const timeCustomTitle = performance.now();
+            console.log("Icon load time: " + (timeCustomTitle - timeStart));
+
+            // tslint:disable-next-line:no-string-literal
```

Comment:
If you want to avoid this, just declare a `ClarityIconElement` interface that extends `HTMLElement` with our few custom properties. It'll be 100 times cleaner and you can use it in the actual code too, allowing you to get proper type checking.